tools/antithesis-go-instrumentor/common/files.go

```diff
@@ -45,22 +47,175 @@ func WriteTextFile(text, file_name string) (err error) {
 	return
 }
 
-func CopyRecursiveNoClobber(from, to string) {
-	commandLine := fmt.Sprintf("cp -n -R %s/* %s", from, to)
+func EnsureDirExists(dirName string) (err error) {
+	var dirInfo os.FileInfo
+	if dirInfo, err = os.Stat(dirName); err != nil {
+		err = os.MkdirAll(dirName, 0777)
+		return
+	}
+
+	// We stat-ed an existing directory or something else
+	if !dirInfo.IsDir() {
+		err = fmt.Errorf("%q is not a directory", dirName)
+	}
+	return
+}
+
+func OpenExistingDir(dirName string) (dir *os.File, err error) {
+	var dirInfo os.FileInfo
+
+	if dir, err = os.Open(dirName); err != nil {
+		dir = nil
+		return
+	}
+
+	if dirInfo, err = dir.Stat(); err != nil {
+		dir.Close()
```

Comment:
should be quietclose() here, and elsewhere that the return value from . Close() is being ignored.